src/terminal/terminal.go

```diff
@@ -0,0 +1,322 @@
+package terminal
+
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
```

Comment:
Might want to rewrite this as something like this if it's mostly influenced by other code:

https://github.com/cert-manager/cert-manager/blob/4c353f1dce4f13b83ca009299f02316116afa12f/pkg/issuer/acme/dns/rfc2136/tsig.go#L36